tinyDA/chain.py

```diff
@@ -217,12 +229,15 @@ def __init__(
         self.posterior_fine = posterior_fine
         self.proposal = proposal
         self.subchain_length = subchain_length
+        self.randomize_subchain_length = randomize_subchain_length 
 
         # set up lists to hold coarse and fine links, as well as acceptance
         # accounting
         self.chain_coarse = []
         self.accepted_coarse = []
         self.is_coarse = []
+        self.promoted_coarse = []
+        self.random_subchain_length = []
```

Comment:
It's a semantic point but, since we are using this list to keep the lengths, whether they are random or not, maybe we can just call this list `self.subchain_lengths`? With an "s" in the end to differentiate it from `self.subchain_length`.